ui/src/websites/wordpress/WordPress.tsx

```diff
@@ -0,0 +1,196 @@
+import { Checkbox, Form, Input, Select } from 'antd';
+import {
+  FileSubmission,
+  WordPressFileOptions,
+  WordPressNotificationOptions,
+  Submission,
+  SubmissionRating
+} from 'postybirb-commons';
+import React from 'react';
+import { WebsiteSectionProps } from '../form-sections/website-form-section.interface';
+import GenericFileSubmissionSection from '../generic/GenericFileSubmissionSection';
+import { GenericSelectProps } from '../generic/GenericSelectProps';
+import GenericSubmissionSection from '../generic/GenericSubmissionSection';
+import { WebsiteImpl } from '../website.base';
+import { LoginDialogProps } from '../interfaces/website.interface';
+import WordPressAccountInfo from './WordPressAccountInfo';
+
+export class WordPress extends WebsiteImpl {
+  internalName: string = 'WordPress';
+  name: string = 'WordPress Instance';
+  supportsAdditionalFiles: boolean = false;
+  supportsTags: boolean = false;
+  loginUrl: string = '';
+
+  LoginDialog = (props: LoginDialogProps) => <WordPressAccountInfo {...props} />;
+
+  FileSubmissionForm = (props: WebsiteSectionProps<FileSubmission, WordPressFileOptions>) => (
+    <WordPressFileSubmissionForm
+      key={props.part.accountId}
+      {...props}
+    />
+  );
+
+  NotificationSubmissionForm = (
+    props: WebsiteSectionProps<Submission, WordPressNotificationOptions>
+  ) => (
+    <WordPressNotificationSubmissionForm
+      key={props.part.accountId}
+      {...props}
+    />
+  );
+}
+
+// TODO: implement options
+// slug (string)
+// comment status (open/close)
+// format (standard, aside, chat, gallery, link, image, quote, status, video, audio)
+// categories (string input)
+// status (publish, future, draft, pending [review], private)
+// sticky (boolean)
+class WordPressNotificationSubmissionForm extends GenericSubmissionSection<
+  WordPressNotificationOptions
+> {
+  renderLeftForm(data: WordPressNotificationOptions) {
+    const elements = super.renderLeftForm(data);
+    elements.push(
+      <div>
+      </div>,
+      <Form.Item label="Post slug">
```

Comment:
q: Can you elaborate on what a slug is in this context and why a user needs to provide one specifically?

Reply:
A slug is the ending part of the URL, it's optional as wordpress will pull it from the title automatically, however this usually looks terrible. So the option to manually provide a slug is provided.

Reply:
eg: https://www.aurawra.org/getting-here

getting-here is the "slug"